jaeger-core/src/main/java/io/jaegertracing/internal/senders/SenderResolver.java

```diff
@@ -56,6 +56,11 @@ public static Sender resolve(Configuration.SenderConfiguration senderConfigurati
     Iterator<SenderFactory> senderFactoryIterator = senderFactoryServiceLoader.iterator();
 
     boolean hasMultipleFactories = false;
+    String requestedFactory = System.getProperty(Configuration.JAEGER_SENDER_FACTORY);
+    boolean isRequestedFactoryAvailable = false;
+    if (!senderFactoryIterator.hasNext()) {
+      log.warn("No sender factories available");
+    }
```

Comment:
You could probably short-circuit here and return `NoopSender()` directly. Doesn't make much difference in the runtime, especially once the VM is warmed up, but makes the code easier to read.